### DISCUSSION TITLE
Queries in loading state with rejected promise

### DISCUSSION THREAD

**Discussion by mwisnicki** [TanStack/query#4164] [Q&A]
I'm using jotai's `atomWithQuery` and when I load the page React's strict mode causes component remount which triggers atom remount which cancels the the query during unmount phase.

The query ends up with rejected promise but in state=loading and never recovers.

1) is cancelation expected and should be followed by retry or does tanstack-query have some mitigation for react's dev mode remounting that is somehow failing to engage here?
2) is the above described state (reject+loading) corrupted or a valid combination?

**Comment by TkDodo** [comment]
assuming you're using v4, look at the `fetchStatus` of the query. If a query is rejected while it is initially loading, it will afterwards be in `fetchStatus: idle`

**Reply by mwisnicki** [reply]
It is indeed idle but what does it mean? It's expected to be stuck?

**Reply by mwisnicki** [reply]
![image](https://user-images.githubusercontent.com/96032/190084429-5d2b226f-dc56-4421-a68a-34257b97f3c8.png)

**Reply by TkDodo** [reply]
please read: https://tanstack.com/query/v4/docs/guides/queries#why-two-different-states

- `loading` means you have no data yet (think "pending" in terms of a promise)
- `idle` means you're currently not fetching any data

if you have a loading query that is fetching data for the first time, and then you abort that fetch, it's neither successful, and it also doesn't have an error, so the only thing that remains is loading / pending.

**Reply by mwisnicki** [reply]
I think the actual problem is that there are no observers. Perhaps a bug in jotai.

**Reply by mwisnicki** [reply]
Reported to Jotai: pmndrs/jotai#1418
